pkg/schema/type.go

```diff
@@ -21,10 +21,10 @@ type Type interface {
 	GetDefinitionPosition() *filepos.Position
 
 	GetDescription() string
-	SetDescription(string)
-	String() string
-	SetTitle(string)
 	GetTitle() string
+	GetExample() (string, interface{})
+	SetDocumentation(documentation)
```

Comment:
This signature will make it not possible for code outside of `schema` to be able to set documentation...

... have you already considered providing setters for `Title` and `Description` and then maybe something like `AddExample(..)`? or `SetExamples(...)`?

Related:
- Today, we can only use the first example in OpenAPI v3.0 output
- But when we turn around and support OpenAPI v3.1 (and that _will_ happen), we'll want to be able to store multiple examples.
- Also, we'd love to be able to support _multiple_ examples in `ytt` Schema, today; no need to limit _that_ part of these system.
- Consider what happens when we push the limitation right up to the seam where it's introduced (i.e. when building the OpenAPI output).